src/App.js

```diff
@@ -21,48 +22,72 @@ const generateSquares = () => {
       currentId += 1;
     }
   }
-
   return squares;
 }
 
 const App = () => {
-
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
 
   // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const updateSquare = (updatedSquare) => {
+    const newSquares = [...squares]
 
+    for(let row of newSquares) {
+      for(let square of row) {
+        if (square.id === updatedSquare.id && square.value === '') {
+          count++
+          square.value = player;          
+        }
+      }
+    };
+  
+    setPlayer(player === PLAYER_1 ? PLAYER_2 : PLAYER_1)
+    setSquares(newSquares);
+
+    if (count > 4) {
+    checkForWinner();
+    }
+  }
 
   const checkForWinner = () => {
     // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if 
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if 
-    //    all three squares have the same value.
 
+    const winPossibilities = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
```

Comment:
This is a clever way to store the winning combinations. You could also consider how you could write loops to check for 3 across in rows and columns in the 2D array.